### ISSUE TITLE
arithmetic_side_effects not triggered when `add_assign` used

### ISSUE THREAD

**Issue by jayvdb** [rust-lang/rust-clippy#12318]
### Summary

Where `foo += 1`  would trigger `arithmetic_side_effects`, the lint can be silenced by using `foo.add_assign(1)`.

Likely this applies to all of the traits in https://doc.rust-lang.org/std/ops/index.html#traits

### Lint Name

arithmetic_side_effects

### Version

```text
rustc 1.76.0 (07dca489a 2024-02-04)
binary: rustc
commit-hash: 07dca489ac2d933c78d3c5158e3f43beefeb02ce
commit-date: 2024-02-04
host: x86_64-unknown-linux-gnu
release: 1.76.0
LLVM version: 17.0.6
```
